File: frontend/src/microservices/flight/objects/FlightBookingPlannerResponse.ts
// 本文件定义 flight 模块的 `FlightBookingPlannerResponse`，作为planner 响应数据并提供 JSON 编解码。

export type FlightBookingPlannerResponse = {
  orderId: string
  orderItemId: string
  status: string
  totalPriceAmount: string
  currency: string
}

export const flightBookingPlannerResponseFromJson = (json: string): FlightBookingPlannerResponse =>
  JSON.parse(json) as FlightBookingPlannerResponse

export const flightBookingPlannerResponseToJson = (value: FlightBookingPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/flight/objects/FlightBookingPlannerResponse.scala
// FlightBookingPlannerResponse defines the flight booking planner response model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightBookingPlannerResponse(
    orderId: String,
    orderItemId: String,
    status: String,
    totalPriceAmount: String,
    currency: String
)

object FlightBookingPlannerResponse:
  given sourceEncoder: Encoder[FlightBookingPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FlightBookingPlannerResponse] = deriveDecoder

